`vscode-extensions.matklad.rust-analyzer` broken on `aarch64-darwin` due to `keytar` dependency missing framework build inputs

### Describe the bug
`vscode-extensions.matklad.rust-analyzer` has been broken on `aarch64-darwin` since #147373 with build log failures that look like (excerpting from <https://hydra.nixos.org/build/161724798/nixlog/2>):

```
...
../src/keytar_mac.cc:1:10: fatal error: 'Security/Security.h' file not found
#include <Security/Security.h>
         ^~~~~~~~~~~~~~~~~~~~~
1 error generated.
make: *** [keytar.target.mk:129: Release/obj.target/keytar/src/keytar_mac.o] Error 1
```

Another patch #149120, fixed a related issue but did not fix the keytar dependency on Darwin.

Fortunately, adding the necessary frameworks as conditional-on-darwin nativeBuildInputs in a manner similar to the way that `lib secret` is added by #149120 fixes things here too.

(PR incoming once I finish writing it up.)

### Notify maintainers
@bkchr, @mweinelt (and @oxalica as a courtesy, though I know she has recently stepped down from maintaining the vscode extension)

--- a/pkgs/misc/vscode-extensions/rust-analyzer/default.nix
+++ b/pkgs/misc/vscode-extensions/rust-analyzer/default.nix
@@ -8,6 +8,8 @@
 , esbuild
 , pkg-config
 , libsecret
+, stdenv
+, darwin
 , setDefaultServerPath ? true
 }:
 
@@ -40,6 +42,9 @@ let
       jq moreutils esbuild
       # Required by `keytar`, which is a dependency of `vsce`.
       pkg-config libsecret
+    ] ++ lib.optionals stdenv.isDarwin [
+      darwin.apple_sdk.frameworks.AppKit
+      darwin.apple_sdk.frameworks.Security
     ];
 
     # Follows https://github.com/rust-analyzer/rust-analyzer/blob/41949748a6123fd6061eb984a47f4fe780525e63/xtask/src/dist.rs#L39-L65

